allennlp/modules/matrix_attention/multi_headed_matrix_attention.py

```diff
@@ -0,0 +1,117 @@
+"""
+A ``Module`` that takes two matrices as input and returns a matrix of attentions.
+"""
+import torch
+from torch.nn import Parameter
+from overrides import overrides
+from allennlp.common import Params
+from allennlp.common.checks import ConfigurationError
+from allennlp.modules.similarity_functions import SimilarityFunction
+from allennlp.modules.matrix_attention.matrix_attention import MatrixAttention
+from allennlp.modules.similarity_functions import DotProductSimilarity
+
+
+@MatrixAttention.register("multiheaded")
+class MultiHeadedMatrixAttention(MatrixAttention):
```

Comment:
This one doesn't match, and so shouldn't be registered like this.  It's not interchangeable with the other `MatrixAttention` classes.

This does raise the question of whether we should combine the attention and the weighted sum, as then we'd have something that's interchangeable here, but let's worry about that later.  I'd probably just remove this file entirely - we already have a multi-headed self-attention encoder that doesn't use this code, and we probably don't need this at all.

Reply:
Why does this one not match? 
It has the same method declaration as MatrixAttention. What am I missing?